asm/istio/istiod-service.yaml

```diff
@@ -42,4 +42,4 @@ spec:
     protocol: TCP
   selector:
     app: istiod
-    istio.io/rev: asm-173-6 # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.rev"}
+    istio.io/rev: ASM_REV # {"$ref":"#/definitions/io.k8s.cli.setters.anthos.servicemesh.rev"}
```

Comment:
one thing I am not sure is if somewhere in our doc uses kubectl apply asm/istio/istiod-service.yaml without kpt cfg set. then the placeholder may break that, but now it is still set to asn-173-6, so i guess we are not using it directly without install_asm. 

Anyway the istiod service fix should be transparent. So let's make this happen.